ddtrace/tracer/span.go

```diff
@@ -88,7 +88,7 @@ type span struct {
 // Context yields the SpanContext for this Span. Note that the return
 // value of Context() is still valid after a call to Finish(). This is
 // called the span context and it is different from Go's context.
-func (s *span) Context() ddtrace.SpanContext { return s.context }
+func (s *span) Context() ddtrace.SpanContextW3C { return s.context }
```

Comment:
I think this makes the change not backwards-compatible, right?

We unfortunately still need the old `Context()` so that API consumers can still get the old 64-bit ID from the span without changing their code.

I thought we were going to implement this as a new optional interface, similar to how Go added functionality to their interfaces in the `database/sql` package: https://pkg.go.dev/database/sql/driver@go1.19.4#ExecerContext only *with* the original `Span` interface as part of the new `SpanW3C` interface. 
i.e.
```
type SpanW3C interface {
    Span
    SpanContextW3C() SpanContextW3C
}
```


Let me know if I'm missing something.